ignite/engine.py

```diff
@@ -0,0 +1,138 @@
+import logging
+from abc import ABCMeta, abstractmethod
+from enum import Enum
+from ignite.history import History
+
+
+class Events(Enum):
+    EPOCH_STARTED = "epoch_started"
+    EPOCH_COMPLETED = "training_epoch_completed"
+    STARTED = "started"
+    COMPLETED = "completed"
+    ITERATION_STARTED = "training_iteration_started"
+    ITERATION_COMPLETED = "training_iteration_completed"
+    EXCEPTION_RAISED = "exception_raised"
+
+
+class Engine(object):
+    __metaclass__ = ABCMeta
+
+    """
+    Abstract Engine class that is the super class of the Trainer and Evaluator engines.
+
+    Parameters
+    ----------
+    process_function : callable
+        A function receiving the current training batch in each iteration, outputing data to be stored in the history
+
+    """
+    def __init__(self, process_function):
+        self._event_handlers = {}
+        self._logger = logging.getLogger(__name__ + "." + self.__class__.__name__)
+        self._logger.addHandler(logging.NullHandler())
+        self._process_function = process_function
+        self.current_iteration = 0
+        self.history = History()
+        self.should_terminate = False
+
+        if self._process_function is None:
+            raise ValueError("Evaluator must have an inference function in order to evaluate")
```

Comment:
The error message is specific to Evaluator currently

Reply:
fixed